sstmac/common/event_scheduler.cc

```diff
@@ -166,6 +176,17 @@ SubComponent::setup()
 #endif
 }
 
+SubComponent::SubComponent(SSTMAC_MAYBE_UNUSED uint32_t id, 
+    const std::string& selfname, SST::Component* parent) :
+#if SSTMAC_INTEGRATED_SST_CORE
+    SST::SubComponent(id),
+    EventScheduler(selfname, 0, parent)
```

Comment:
I will need to understand this better. The (undocumented) intent is that EventScheduler is never used with core. I would prefer to avoid this inheritance pattern. What was the reason for doing this? (so I can think about the best way)

Reply:
Same as above.